backend/src/routes/api/resources.js

```diff
@@ -0,0 +1,118 @@
+import { Router } from "express";
+import auth from "../../middleware/firebaseAuth";
+import scenarioAuth from "../../middleware/scenarioAuth";
+
+import {
+  createResource,
+  getResourceById,
+  deleteResourceById,
+  addFlag,
+  removeFlag,
+  getAllVisibleResources,
+  updateResourceById,
+} from "../../db/daos/resourcesDao";
+
+const router = Router();
+
+const HTTP_OK = 200;
+const HTTP_CREATED = 201;
+const HTTP_NO_CONTENT = 204;
+const HTTP_BAD_REQUEST = 400;
+const HTTP_NOT_FOUND = 404;
+const HTTP_FORBIDDEN = 403;
+
+// Apply auth middleware to all routes below this point
+router.use(auth);
+
+// Create a New Resource
+router.post("/", async (req, res) => {
+  const { type, content, name } = req.body;
+  if (!type || !content || !name) {
+    return res.status(HTTP_BAD_REQUEST).send("Bad Request");
+  }
+  const newResource = await createResource(type, content, name);
```

Comment:
Potential for uncaught server error on invalid type? (e.g. when type is `ABCDEFG`?

Reply:
invalid types caught in createResource function in the DAO:

![image](https://github.com/user-attachments/assets/2634d22e-6151-4078-a1aa-bc717528f448)

Reply:
I would recommend using the error handling function and custom error class I created, so that the errors get sent back in the response. All you need to do is wrap the function using the handle() and throw HttpError instead.